### PR TITLE
Jetpack Manage: Implement pricing summary in the review licenses modal

### PR DESCRIPTION
**Pull request by yashwin**
Resolves https://github.com/Automattic/jetpack-genesis/issues/98

## Proposed Changes

This PR displays the pricing summary in the review licenses modal.

## Testing Instructions

**Prerequisites**

Since these changes are made specifically for agencies, you must set yourself(partner) as an agency - 2c49b-pb. Make sure to switch it back to the previous type.

**Instructions**

1. Visit the Jetpack Cloud link > Replace `dashboard` in the URL with `/partner-portal/issue-license?flags=jetpack/bundle-licensing`.
2. Select a few licenses and click the `Review X licenses` button > Verify that the total price is shown along with the `Issue X license` button on the right side of the modal. Please note the CTA & Learn more link doesn't do anything. It will be updated in upcoming PRs. The pricing breakdown will the done in another PR. The discounted price needs to be updated based on the selected bundle(TODO)

Large screen:

Screenshot 2023-11-24 at 11 11 06 AM

Tablet view:

Screenshot 2023-11-24 at 11 24 21 AM

Mobile view:

Screenshot 2023-11-24 at 11 12 01 AM

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/jetpack-cloud/sections/partner-portal/issue-license-v2/lib/pricing.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+			Object.values( userProducts ).find(
+				( p ) =>
+					p.billing_product_slug === yearlyProduct.billing_product_slug &&
+					p.product_term === 'month'
+			);
+		// If a monthly product is found, calculate the actual cost and discount percentage
+		if ( monthlyProduct ) {
+			const actualCost = isDailyPricing ? monthlyProduct.cost / 365 : monthlyProduct.cost;
+			const discountedCost = actualCost - productCost;
+			discountInfo.discountPercentage = ! productCost
```
Comment (jkguidaven) [inline client/jetpack-cloud/sections/partner-portal/issue-license-v2/lib/pricing.ts]: [non-blocker] We can convert this to a positive condition.

```
discountInfo.discountPercentage = productCost
   ? Math.round( ( discountedCost / actualCost ) * 100 ) : 100;
```

File: client/jetpack-cloud/sections/partner-portal/issue-license-v2/review-licenses/pricing-summary.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+	}, [] );
+
+	const link = '#link'; // TODO: Add link
+
+	const currency = selectedLicenses[ 0 ].currency; // FIXME: Fix if multiple currencies are supported
+
+	return (
+		<>
+			<div className="review-licenses__pricing">
+				<span className="review-licenses__price">
```
Comment (jkguidaven) [inline client/jetpack-cloud/sections/partner-portal/issue-license-v2/review-licenses/pricing-summary.tsx]: [non-blocker]I suggest to name this `review-licenses__pricing-discounted` and the other `review-licenses__pricing-original` to match the corresponding content.

**Review by jkguidaven — APPROVED** [review]
I confirmed that the price and CTA are properly rendered across different screen sizes. I left a few non-blocking comments, but otherwise everything is good. Nice work! 👍

**Comment by yashwin** [comment]
Thanks for the review & feedback. I have addressed your comments: https://github.com/Automattic/wp-calypso/pull/84496/commits/e298ae38641518c7250bf2e4f780c4f1d8767bf8. Could you please take another look? TIA 🙂

**Review by jkguidaven — APPROVED** [review]
Thanks for addressing my comments. Retested and works as expected. The new changes LGTM 👍

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/9765006

Thank you @yashwin for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/9765006).

**Pull request merged**